HHS/Head-Start-TTADP
# [TTAHUB-1360] AR form, don't save on navigation when form values haven't changed

## Description of change

Different approach to this problem. In this solution, the `findWhatsChanged` helper has a small change where endState and startDate, when they are not valid dates, are removed from the data to save instead of setting them to null. This is because when nothing on the page changes, these are usually the only two values that are returned by `findWhatsChanged`.

When `findWhatsChanged` returns an empty object, we don't send the PUT request. The rest of the `onSave` logic still happens as usual.


## How to test

Keeping your network tab open, create an AR or edit an existing one. Ensure that PUT requests are only sent when navigating between AR form sections if you've actually changed the value of some field.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1360


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
